src/ConnectedMode/Binding/IRoslynSettingsFileSynchronizer.cs

```diff
@@ -0,0 +1,35 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Threading.Tasks;
+
+namespace SonarLint.VisualStudio.ConnectedMode.Binding
+{
+    /// <summary>
+    /// Responsible for listening to <see cref="IServerIssuesStore.ServerIssuesChanged"/> and calling
+    /// <see cref="IRoslynSettingsFileStorage.Update"/> with the new suppressions.
+    /// </summary>
+    public interface IRoslynSettingsFileSynchronizer : IDisposable
```

Comment:
I had to move the interface out of `RoslynSuppressions` so that the migration code can import it.

Normally we'd put shared interfaces like this in `Core`. However, `Core` is starting to become a bit of a dumping ground for interfaces to multiple different unrelated features.

So, since the Roslyn suppressions file is only relevant in Connected Mode I decided to move it to the `ConnectedMode` assembly instead. `RoslynSuppressions` already has a project dependency on the `ConnectedMode` project.

If we need to move the interfaces in the future we can easily do so.